src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoCommon.cs

```diff
@@ -122,5 +144,14 @@ private IEnumerable<TRuntimeProperty> CreateGenericIEnumerableFromList(IList sou
                 yield return (TRuntimeProperty)item;
             }
         }
+
+        private IEnumerable<KeyValuePair<string, TRuntimeProperty>> CreateGenericIEnumerableFromDictionary(IDictionary sourceDictionary)
+        {
+            foreach (object item in sourceDictionary)
```

Comment:
You can just put `foreach (DictionaryEntry item in sourceDictionary)` here. Foreach can do some spooky stuff.